extensions/iam/decentralized-identity/identity-did-crypto/src/main/java/org/eclipse/dataspaceconnector/iam/did/crypto/credentials/VerifiableCredentialFactory.java

```diff
@@ -121,61 +86,48 @@ public static SignedJWT create(PrivateKeyWrapper privateKey, Map<String, String>
     /**
      * Verifies a VerifiableCredential using the issuer's public key
      *
-     * @param verifiableCredential a {@link SignedJWT} that was sent by the claiming party.
-     * @param publicKey The claiming party's public key
-     * @return true if verified, false otherwise
-     */
-    public static boolean verify(SignedJWT verifiableCredential, ECKey publicKey) {
-        try {
-            return verifiableCredential.verify(new ECDSAVerifier(publicKey));
-        } catch (JOSEException e) {
-            throw new CryptoException(e);
-        }
-    }
-
-    /**
-     * Verifies a VerifiableCredential using the issuer's public key
-     *
-     * @param verifiableCredential a {@link SignedJWT} that was sent by the claiming party.
+     * @param jwt       a {@link SignedJWT} that was sent by the claiming party.
      * @param publicKey The claiming party's public key, passed as a {@link PublicKeyWrapper}
+     * @param audience  The intended audience
      * @return true if verified, false otherwise
      */
-    public static boolean verify(SignedJWT verifiableCredential, PublicKeyWrapper publicKey) {
+    public static Result<Void> verify(SignedJWT jwt, PublicKeyWrapper publicKey, String audience) {
+        // verify JWT signature
         try {
-            return verifiableCredential.verify(publicKey.verifier());
+            var verified = jwt.verify(publicKey.verifier());
+            if (!verified) {
+                return Result.failure("Invalid signature");
+            }
         } catch (JOSEException e) {
-            throw new CryptoException(e);
+            // Unable to verify, e.g. the JWS algorithm is not supported
+            return Result.failure(e.getMessage());
         }
-    }
 
-    /**
-     * Verifies a VerifiableCredential using the issuer's public key
-     *
-     * @param verifiableCredential a {@link SignedJWT} that was sent by the claiming party.
-     * @param publicKeyPemContent The claiming party's public key, i.e. the contents of the public key PEM file.
-     * @return true if verified, false otherwise
-     */
-    public static boolean verify(SignedJWT verifiableCredential, String publicKeyPemContent) {
+        JWTClaimsSet jwtClaimsSet;
         try {
-            var key = ECKey.parseFromPEMEncodedObjects(publicKeyPemContent);
-            return verify(verifiableCredential, (ECKey) key);
-        } catch (JOSEException e) {
-            throw new CryptoException(e);
+            jwtClaimsSet = jwt.getJWTClaimsSet();
+        } catch (ParseException e) {
+            // The payload of the JWT doesn't represent a valid JSON object and a JWT claims set
```

Comment:
same as above

Reply:
This is already in the message produced by nimbus, e.g. `"Payload of JWS object is not a valid JSON object"`